docs/fastdds/xtypes/type_serializing.rst

```diff
@@ -0,0 +1,213 @@
+            :language: json
+
+Sequence types
+""""""""""""""
+
+Sequence types are used to represent ordered collections of elements, similar to arrays but with
+dynamic length.
+Below is an example of the definition of sequence types in IDL:
+
+.. literalinclude:: /../code/DynamicTypesIDLExamples.idl
+    :language: omg-idl
+    :start-after: //!--IDL_SEQUENCES
+    :end-before: //!--
+
+The previous `DynamicData` object would be serialized as follows:
+
+.. literalinclude:: /../code/json/Sequences.json
+    :language: json
+
+Array types
+"""""""""""
+
+Array types are used to represent fixed-size collections of elements.
+The following example shows the definition of array types in IDL:
+
+.. literalinclude:: /../code/DynamicTypesIDLExamples.idl
+    :language: omg-idl
+    :start-after: //!--IDL_ARRAYS_JSON
+    :end-before: //!--
+
+The previous `DynamicData` object would be serialized as follows:
+
+.. literalinclude:: /../code/json/Arrays.json
+    :language: json
+
+Map Types
+"""""""""
+
+Map types represent collections of key-value pairs, allowing for the efficient lookup of values based
+on unique keys.
+Below is an example of the definition of map types in IDL:
+
+.. literalinclude:: /../code/DynamicTypesIDLExamples.idl
+    :language: omg-idl
+    :start-after: //!--IDL_MAPS
+    :end-before: //!--
+
+The previous `DynamicData` object would be serialized as follows:
+
+.. literalinclude:: /../code/json/Maps.json
+    :language: json
+
+Structure Types
+"""""""""""""""
+
+Structure types are used to group different types of data together.
+Here is an example of the definition of structure types in IDL:
+
+.. literalinclude:: /../code/DynamicTypesIDLExamples.idl
+    :language: omg-idl
+    :start-after: //!--IDL_STRUCT
+    :end-before: //!--
+
+The previous `DynamicData` object would be serialized as follows:
+
+.. literalinclude:: /../code/json/Structs.json
+    :language: json
+
+Union Types
+"""""""""""
+
+Union types are a special type of structure type where only one member exists.
+Below is an example of the definition of union types in IDL:
+
+.. literalinclude:: /../code/DynamicTypesIDLExamples.idl
+    :language: omg-idl
+    :start-after: //!--IDL_UNION
+    :end-before: //!--
+
+The previous `DynamicData` object would be serialized as follows:
+
+.. literalinclude:: /../code/json/Unions.json
+    :language: json
+
+Example: Remote discovered types introspection
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The following code demonstrates how to implement remote type introspection using FastDDS in C++.
+This feature allows a subscriber to introspect the remotely discovered type and serialize it into a more manageable
+and understandable format.
+Once a type is discovered, it can be registered and every time the subscriber receives new data related to this type,
+the corresponding DynamicData can be obtained from the DynamicDataFactory and serialized into a JSON string format.
```

Comment:
The example should be focused on how to use the ``json_serialize`` function and not on how to implement type introspection. Also, add a link to the type introspection example just for completeness.